src/V10/ResourceNames.cs

```diff
@@ -80,7 +83,7 @@ public static string AdGroupAd(long customerId, long adGroupId, long adId)
             AssetFieldType fieldType)
         {
             return $"customers/{customerId}/adGroupAssets/{adGroupId}~{assetId}~" +
-                $"{fieldType.ToString().ToUpper()}";
+                $"{ProtobufUtilities.GetOriginalEnumFieldName(fieldType)}";
```

Comment:
Do you have tests for special cases like this?

Reply:
Added tests.